packages/collector/test/tracing/cloud/aws-sdk/v3/kinesis/app.js

```diff
@@ -0,0 +1,216 @@
+};
+const operationNames = Object.keys(availableOperations);
+const commandMapping = {
+  createStream: CreateStreamCommand,
+  putRecord: PutRecordCommand,
+  putRecords: PutRecordsCommand,
+  getRecords: GetRecordsCommand,
+  getShardIterator: GetShardIteratorCommand,
+  listStreams: ListStreamsCommand,
+  listShards: ListShardsCommand
+};
+function enforceErrors(options) {
+  // this will enforce error for all commands except liststreams
+  options.ShardCount = 0;
+}
+async function executeOperation(operation, withError, method) {
+  const operationOptions = availableOperations[operation];
+  const mergedOptions = { ...operationOptions };
+
+  if (withError) {
+    enforceErrors(mergedOptions);
+  }
+  const command = commandMapping[operation];
+  switch (operation) {
+    case 'getRecords': {
+      const data = await executeOperation('getShardIterator', withError, method);
+      mergedOptions.ShardIterator = data.ShardIterator;
+      return kinesisSend(new command(mergedOptions), method);
+    }
+    case 'getShardIterator': {
+      const shards = await executeOperation('listShards', withError, method);
+      const { ShardId, SequenceNumberRange } = shards.Shards[0];
+      const StartingSequenceNumber = SequenceNumberRange.StartingSequenceNumber;
+      mergedOptions.ShardId = ShardId;
+      mergedOptions.StartingSequenceNumber = StartingSequenceNumber;
+      return kinesisSend(new command(mergedOptions), method);
+    }
+    default:
+      return kinesisSend(new command(mergedOptions), method);
+  }
+}
+async function kinesisSendWithPromise(command) {
+  return kinesis.send(command).then(result => {
+    return result;
+  });
+}
+
+function kinesisSendWithCB(command, cb) {
+  kinesis.send(command, (error, result) => {
+    if (error) {
+      cb(new Error(`Error executing "${command.constructor.name}": ${error.message}`));
+    } else {
+      cb(null, result);
+    }
+  });
+}
+
+function kinesisV2SendWithPromise(command) {
+  return new Promise((resolve, reject) => {
+    kinesisV2.send(command, (error, result) => {
+      if (error) {
+        reject(new Error(`Error executing "${command.constructor.name}": ${error.message}`));
+      } else {
+        resolve(result);
+      }
+    });
+  });
+}
+
+function kinesisSendV2WithCB(command, callback) {
+  kinesisV2.send(command, (error, result) => {
+    if (error) {
+      callback(new Error(`Error executing "${command.constructor.name}": ${error.message}`));
+    } else {
+      callback(null, result);
+    }
+  });
+}
+async function kinesisSend(command, method) {
+  switch (method) {
+    case 'async':
+      return kinesis.send(command);
+    case 'promise':
+      return kinesisSendWithPromise(command);
+    case 'cb':
+      return new Promise((resolve, reject) => {
+        kinesisSendWithCB(command, (err, result) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(result);
+          }
+        });
+      });
+    case 'async-v2':
```

Comment:
Suggestion: rename all of the v2 strings in this file to v3.

Reply:
This case is to support v2 compatible style in v3.

Reply:
Ah ok then nvm :)